### ISSUE TITLE
Extend 'CommandAttribute' with 'IsOptional'property.

### ISSUE THREAD

**Issue by MarcinIN** [inausoft/netCLI#13]
After trying v1.0.0 and my reply to @matiduck that user can throw custom exception from within `handler` I've came to conclusion that having global `try` block I cannot differentiate if `ArgumentException` was thrown because of some key information of `command` were missing or something went wrong in another part of application.

Perhaps `option` must be extended with `Required` property so that deserializer can throw custom exception.

```cs
...

try
{
    return Executor.RunCLI(config, args, handlers);
}
catch(ArgumentException ex)
{
    //How to inform user that required option was not specified? Not able to distinguish between both of exceptions..
}

...

[Command("delete")]
class DeleteCommand
{
    [Option("path", "File path.")]
    public string From { get; set; }
}

class DeleteCommandHandler : CommandHandler<DeleteCommand>
{
    public string SomeConfig { get; set; }

    public override int Run(DeleteCommand options)
    {
        if(options.Path == null)
        {
             throw new ArgumentException("Path");
        }

        SomeMethod(SomeConfig);

        return 0;
    }

    public void SomeMethod(string config)
    {
        if(SomeConfig == null)
        {
             throw new ArgumentException("Config");
        }
        ...
    }
}

```

**Comment by MarcinIN** [comment]
@matiduck @jmarlew @xHoody any insight on the approach?

```cs
[Command("delete")]
class DeleteCommand
{
    [Option("path", "File path.", true)]
    public string From { get; set; }
}
```
or
```cs
[Command("delete")]
class DeleteCommand
{
    [Option("path", "File path.")]
    [OptionRequired()]
    public string From { get; set; }
}
```
or
```cs
[Command("delete")]
class DeleteCommand
{
    [Option("path")]
    [OptionRequired()]
    [OptionDescription("File path.")]
    public string From { get; set; }
}
```

**Comment by jmarlew** [comment]
Another option for option (sic!) syntax could be:
```cs
[Option("path", "File path.", IsRequired = true)]
```

**Comment by MarcinIN** [comment]
Let's go with:
```cs
[Option("path", "File path.", IsOptional = true)]
```
How to define it with `=` syntax instead of `:` ?

**Comment by jmarlew** [comment]
See https://docs.microsoft.com/pl-pl/dotnet/csharp/language-reference/language-specification/attributes#positional-and-named-parameters

**Comment by MarcinIN** [comment]
Is available in the 2.0.0-rc.1
